google_fonts/lib/src/google_fonts_base.dart

```diff
@@ -125,3 +125,63 @@ String _fontStyleString(FontStyle fontstyle) {
   }
   return '';
 }
+
+class GoogleFontsFamily {
+  const GoogleFontsFamily(this.fontWeight, this.fontStyle, [this.url]);
+
+  final int fontWeight;
+  final FontStyle fontStyle;
+  final String url;
+
+  // From a string key, for example, `500italic`.
+  GoogleFontsFamily.fromString(String key, [this.url])
+      : this.fontWeight = key == _regular || key == _italic
+            ? 400
+            : int.parse(key.replaceAll(_italic, '')),
+        this.fontStyle =
+            key.contains(_italic) ? FontStyle.italic : FontStyle.normal;
+
+  GoogleFontsFamily.fromTextStyle(TextStyle style, [this.url])
+      : this.fontWeight = (style?.fontWeight?.index ?? 4) * 100,
```

Comment:
If this is 0 for FontWeight.w100, do we need to add 1?

Reply:
Good catch, I got around this by making `GoogleFontsFamily.fontWeight` _always_ represent the 0-based index value of the font weight.